### PR TITLE
fix - duplicates form and query params

### PR DESCRIPTION
**Pull request by viclovsky**

### CODE REVIEW

**Comment by eroshenkoam** [comment]
test this please

File: rarc-core/src/main/java/ru/lanwen/raml/rarc/api/FormQueryParamsMerge.java [hunk 1]
```diff
@@ -49,20 +48,22 @@ public FormQueryParamsMerge(ApiResourceClass apiResourceClass) {
...
                     list.add(new AddAnyParamMethod(((AddFormParamMethod) elem).getParam(), elem.name(), ((AddFormParamMethod) elem).getReq(), apiResourceClass));
-                } else {
-                    list.add(elem);
+                    return;
                 }
-            } else {
+            }
+
+            // Check duplicates
+            Optional<AddParamMethod> alreadyIn = list.stream().filter(item -> item instanceof AddAnyParamMethod).findFirst();
```
Comment (lanwen) [inline rarc-core/src/main/java/ru/lanwen/raml/rarc/api/FormQueryParamsMerge.java]: `boolean alreadyIn = list.stream().anyMatch(AddAnyParamMethod.class::isInstance)`

File: rarc-example/src/test/java/ru/lanwen/raml/test/ApiExampleUsageTest.java [hunk 2]
```diff
@@ -26,4 +26,15 @@ public void shouldBeAbleToCompile() throws Exception {
...
     }
+
+    @Test
+    public void shouldUseDuplicateParams() {
+        ApiExample.example(
+                ApiExample.Config.exampleConfig()
+                        .withReqSpecSupplier(
+                                () -> new RequestSpecBuilder().setBaseUri("http://your_host/")
+                        )
+        ).hardDuplicate().withDuplicatedParam("blah").post(identity()).prettyPeek();
```
Comment (lanwen) [inline rarc-example/src/test/java/ru/lanwen/raml/test/ApiExampleUsageTest.java:37]: don't do real post - or ignore the test

**Review by lanwen — CHANGES_REQUESTED** [review]

File: rarc-example/src/test/java/ru/lanwen/raml/test/ApiExampleUsageTest.java (hunk 2 above)
Comment (viclovsky) [reply rarc-example/src/test/java/ru/lanwen/raml/test/ApiExampleUsageTest.java:37]: Don't need to ignore test cause the whole class has tag @ignore

File: rarc-core/src/main/java/ru/lanwen/raml/rarc/api/FormQueryParamsMerge.java (hunk 1 above)
Comment (viclovsky) [reply rarc-core/src/main/java/ru/lanwen/raml/rarc/api/FormQueryParamsMerge.java]: ok

File: rarc-example/src/test/java/ru/lanwen/raml/test/ApiExampleUsageTest.java (hunk 2 above)
Comment (lanwen) [reply rarc-example/src/test/java/ru/lanwen/raml/test/ApiExampleUsageTest.java:37]: ah, ok

**Review by lanwen — APPROVED** [review]

**Pull request merged**